snapcraft/internal/repo.py

```diff
@@ -299,7 +299,10 @@ def _get(self, apt_cache, package_names):
                 logger.debug(
                     'Marking {!r} (and its dependencies) to be fetched'.format(
                         name))
-                apt_cache[name].mark_install()
+                version = None
+                name_arch, version = _get_pkg_name_parts(name)
+                _set_candidate(apt_cache[name_arch], version)
```

Comment:
Let me know what you think of my update.  It's just a minor refactor to get complexity down not a major overhaul.